cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/task/SampleLoadingTask.java

```diff
@@ -33,8 +39,14 @@ public void run() {
                                                             _brokerMetricSampleAggregator));
       ModelParameters.updateModelCoefficient();
     } finally {
-      // The sample loading task is run before the load monitor starts.
-      _loadMonitorTaskRunner.setState(LoadMonitorTaskRunner.LoadMonitorTaskRunnerState.RUNNING);
+      // If Cruise Control has triggered operations before loading is finished, transfer monitor state to paused state;
```

Comment:
Isn't there a race condition here? -- i.e. in general, if we would like to avoid such race conditions, shouldn't we update executor state and load monitor state atomically?

Reply:
Wrap this part of logic with `synchronized (_loadMonitorTaskRunner)`